pkg/core/dao/dao.go

```diff
@@ -163,7 +164,13 @@ func (dao *Simple) GetContractState(hash util.Uint160) (*state.Contract, error)
 // PutContractState puts given contract state into the given store.
 func (dao *Simple) PutContractState(cs *state.Contract) error {
 	key := storage.AppendPrefix(storage.STContract, cs.ScriptHash().BytesBE())
-	return dao.Put(cs, key)
+	if err := dao.Put(cs, key); err != nil {
+		return err
+	}
+	key = key[:5]
```

Comment:
I think it is better to make this a separate `dao.PutContractScriptHash` method. Hash is needed to be updated only during migration, not upon every `PutContractState`

Reply:
We should also *put* hash into the storage while creating/adding new contract. Each time we add new contract to the storage, the pair `hash-ID` should also be stored. We should also update this pair each time contract hash is updated. As a result, the **only** place in **the whole project** when we don't have to update `hash-ID` pair is when we [update contract manifest during migration](https://github.com/nspcc-dev/neo-go/blob/master/pkg/core/interop_neo.go#L161) (which is quite rear event).

If we make a separate `dao.PutContractScriptHash`  method, we'll have to call it almost each time after `dao.PutContractState`. So do we really want to split this methods?

I also thought about changing the name of `dao.PutContractState` to `dao.PutContractStateAndID` or smth similar.

Reply:
Ok, but it still can be a separate function (may be private).